### PR TITLE
Address accessibility-related ESLint issues

### PR DESCRIPTION
**Pull request by Gelio**
This PR addresses issues found by ESLint after enabling https://github.com/jsx-eslint/eslint-plugin-jsx-a11y

I have also sneaked in 1 TS improvement (define an expected global property)

This PR depends on #165

### CODE REVIEW

File: tdesign/src/Link/MuiLink.tsx [hunk 1]
```diff
@@ -47,6 +47,8 @@ export const NextLinkComposed = forwardRef<
       passHref={passHref}
       locale={locale}
     >
+      {/* SAFETY: `children` are spread in `other` */}
+      {/* eslint-disable-next-line jsx-a11y/anchor-has-content */}
       <a ref={ref} {...(other as any)} />
```
Comment (milesrichardson) [inline tdesign/src/Link/MuiLink.tsx:52]: FYI, this same code is the source of the mess in the marketing repo (copied from here). This needs to be updated to use new MUI recommended best practices. Also note the `any` came here due to profiling revealing it to significantly increase typechecking speed in files that imported `MuiLink`

File: tdesign/src/Layout/BaseLayout/BaseLayout.tsx [hunk 2]
```diff
@@ -123,6 +123,8 @@ const BaseLayout = ({
             <UserThemeContext.Consumer>
               {({ userColors, setUserColors, toggleDarkMode }) => (
                 <div>
+                  {/* TODO: fix me */}
+                  {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */}
```
Comment (milesrichardson) [inline tdesign/src/Layout/BaseLayout/BaseLayout.tsx:132]: (note this is a hidden toggle atm anyway)

**Review by milesrichardson — APPROVED** [review]

**Comment by milesrichardson** [comment]
LGTM.

File: tdesign/src/Link/MuiLink.tsx (hunk 1 above)
Comment (Gelio) [reply tdesign/src/Link/MuiLink.tsx:52]: I remember the part about the `any` annotation to opt-out of typechecking. What is the other "mess" you're referring to? This component as a whole :smile: ?

File: tdesign/src/Layout/BaseLayout/BaseLayout.tsx (hunk 2 above)
Comment (Gelio) [reply tdesign/src/Layout/BaseLayout/BaseLayout.tsx:132]: That makes me feel more calm about not fixing it right away :smile:

**Comment by Gelio** [comment]
Thanks for the review! I'll merge the PR

**Pull request merged**